### ISSUE TITLE
Unable to upgrade due to click-8.1.3 not supported for interpreter python 2.7

### ISSUE THREAD

**Issue by j-hui** [NixOS/nixpkgs#203198]
### Describe the bug

I'm runnig NixOS and am trying to upgrade to 22.11 channel. However, I get this error message:

```console
$ sudo nixos-rebuild switch
building Nix...
building the system configuration...
error: click-8.1.3 not supported for interpreter python2.7
```

The trace does not appear to be helpful. I don't direclty depend on `click` so I'm not sure what's causing my build to fail.

### Steps To Reproduce
1. ```sudo nix-channel --add https://nixos.org/channels/nixos-22.11 nixos && sudo nix-channel --update```
2. ```sudo nixos-rebuilt switch```

### Expected behavior
Upgrade my system without errors.

### Notify maintainers

@SuperSandro2000 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.111, NixOS, 22.05 (Quokka), 22.05.2199.478f3cbc844`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(root): `"nixos-22.11, nixpkgs-unstable"`
 - channels(j-hui): `"nixpkgs-unstable-22.05pre344968.8928525bd8b"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 ```

**Comment by freezeboy** [comment]
Hi you may be using `pythonPackages.click` in your configuration, due to a migration to `click` version  `8.1.3`, it is now only available for `python3`

**Comment by j-hui** [comment]
@freezeboy as I mentioned, I am not directly using `click` in my configuration. I suspect it’s a dependency for something else I use but I’m not sure what.

**Comment by j-hui** [comment]
Also, I’m not trying to install it with python2 anywhere. So why does it think I’m trying to use it with python2?

**Comment by freezeboy** [comment]
Oops I missed that part

You can try to see the dependency tree using (if you use flakes):
```
nix --experimental-features nix-command --experimental-features flake why-depends .#nixosConfigurations.hostname.config.system.build.toplevel .#python2Packages.click
```
I don't know the equivalent for classic expressions

**Comment by j-hui** [comment]
Ah `why-depends` was what I was looking for! It seems like there is a classical equivalent for that command, I just didn't know it existed. (Actually I just grepped through the result of `nix-store -q --tree /run/current-system`, but I wouldn't have found that if I didn't know to Google "why-depends").

The turns out offending package was `pythonPackages.compiledb`, which depends on `click`. I forgot that `pythonPackages` defaults to Python 2... I fixed it just by changing the package to `python3Packages.compiledb`.

Closing this issue now.
